File: TP2PARTIE2/app/src/main/java/umontpellier/gl1/tp2partie2/DatabaseHelper.java
package umontpellier.gl1.tp2partie2;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

public class DatabaseHelper extends SQLiteOpenHelper {

    private static final String DATABASE_NAME = "UserManager.db";
    private static final int DATABASE_VERSION = 1;
    private static final String TABLE_USER = "user";
    private static final String COLUMN_USER_ID = "user_id";
    private static final String COLUMN_USER_NAME = "user_name";
    private static final String COLUMN_USER_EMAIL = "user_email";
    private static final String COLUMN_USER_PASSWORD = "user_password";


    // Commande SQL pour créer la table
    private static final String CREATE_TABLE_USER = "CREATE TABLE " + TABLE_USER + "("
            + COLUMN_USER_ID + " INTEGER PRIMARY KEY AUTOINCREMENT,"
            + COLUMN_USER_NAME + " TEXT,"
            + COLUMN_USER_EMAIL + " TEXT,"
            + COLUMN_USER_PASSWORD + " TEXT" + ")"; // Ajout de la colonne pour l'image

    public DatabaseHelper(Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL(CREATE_TABLE_USER);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        db.execSQL("DROP TABLE IF EXISTS " + TABLE_USER);
        onCreate(db);
    }

    // Méthode pour ajouter un utilisateur avec une image
    public void addUser(String name, String email, String password) {
        SQLiteDatabase db = this.getWritableDatabase();
        ContentValues values = new ContentValues();
        values.put(COLUMN_USER_NAME, name);
        values.put(COLUMN_USER_EMAIL, email);
        values.put(COLUMN_USER_PASSWORD, password);
        db.insert(TABLE_USER, null, values);
        db.close();
    }

    // Méthode pour vérifier si l'email existe déjà
    public boolean checkUser(String email) {
        SQLiteDatabase db = this.getReadableDatabase();
        Cursor cursor = db.query(TABLE_USER, new String[]{COLUMN_USER_ID}, COLUMN_USER_EMAIL + "=?", new String[]{email}, null, null, null);
        int cursorCount = cursor.getCount();
        cursor.close();
        db.close();
        return cursorCount > 0;
    }

    // Méthode pour vérifier l'email et le mot de passe
    public boolean checkUser(String email, String password) {
        SQLiteDatabase db = this.getReadableDatabase();
        Cursor cursor = db.query(TABLE_USER, new String[]{COLUMN_USER_ID}, COLUMN_USER_EMAIL + "=? AND " + COLUMN_USER_PASSWORD + "=?", new String[]{email, password}, null, null, null);
        int cursorCount = cursor.getCount();
        cursor.close();
        db.close();
        return cursorCount > 0;
    }


    public String getUserName(String email) {
        SQLiteDatabase db = this.getReadableDatabase();
        Cursor cursor = db.query(TABLE_USER, new String[]{COLUMN_USER_NAME}, COLUMN_USER_EMAIL + "=?", new String[]{email}, null, null, null);
        if (cursor != null && cursor.moveToFirst()) {
            int columnIndex = cursor.getColumnIndex(COLUMN_USER_NAME);
            if (columnIndex >= 0) { // Vérifiez que la colonne existe
                String userName = cursor.getString(columnIndex);
                cursor.close();
                db.close();
                return userName;
            }
        }
        if (cursor != null) {
            cursor.close();
        }
        db.close();
        return null;
    }





}

File: TP2App1/app/src/main/java/umontpellier/gl1/tp2app1/DatabaseHelper.kt
package umontpellier.gl1.tp2app1

import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

class DatabaseHelper(context: Context) : SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {

    companion object {
        private const val DATABASE_NAME = "UserManager.db"
        private const val DATABASE_VERSION = 1
        private const val TABLE_USER = "user"
        private const val COLUMN_USER_ID = "user_id"
        private const val COLUMN_USER_NAME = "user_name"
        private const val COLUMN_USER_EMAIL = "user_email"
        private const val COLUMN_USER_PASSWORD = "user_password"

        // Commande SQL pour créer la table
        private const val CREATE_TABLE_USER = ("CREATE TABLE $TABLE_USER("
                + "$COLUMN_USER_ID INTEGER PRIMARY KEY AUTOINCREMENT,"
                + "$COLUMN_USER_NAME TEXT,"
                + "$COLUMN_USER_EMAIL TEXT,"
                + "$COLUMN_USER_PASSWORD TEXT)")
    }

    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL(CREATE_TABLE_USER)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        db.execSQL("DROP TABLE IF EXISTS $TABLE_USER")
        onCreate(db)
    }

    // Méthode pour ajouter un utilisateur
    fun addUser(name: String, email: String, password: String) {
        val db = this.writableDatabase
        val values = ContentValues().apply {
            put(COLUMN_USER_NAME, name)
            put(COLUMN_USER_EMAIL, email)
            put(COLUMN_USER_PASSWORD, password)
        }
        db.insert(TABLE_USER, null, values)
        db.close()
    }

    // Méthode pour vérifier si l'email existe déjà
    fun checkUser(email: String): Boolean {
        val db = this.readableDatabase
        val cursor: Cursor = db.query(
            TABLE_USER, arrayOf(COLUMN_USER_ID), "$COLUMN_USER_EMAIL=?", arrayOf(email),
            null, null, null
        )
        val cursorCount = cursor.count
        cursor.close()
        db.close()
        return cursorCount > 0
    }

    // Méthode pour vérifier l'email et le mot de passe
    fun checkUser(email: String, password: String): Boolean {
        val db = this.readableDatabase
        val cursor: Cursor = db.query(
            TABLE_USER, arrayOf(COLUMN_USER_ID),
            "$COLUMN_USER_EMAIL=? AND $COLUMN_USER_PASSWORD=?", arrayOf(email, password),
            null, null, null
        )
        val cursorCount = cursor.count
        cursor.close()
        db.close()
        return cursorCount > 0
    }

    // Méthode pour récupérer le nom de l'utilisateur
    fun getUserName(email: String): String? {
        val db = this.readableDatabase
        val cursor: Cursor? = db.query(
            TABLE_USER, arrayOf(COLUMN_USER_NAME), "$COLUMN_USER_EMAIL=?", arrayOf(email),
            null, null, null
        )
        return cursor?.use {
            if (it.moveToFirst()) {
                val columnIndex = it.getColumnIndex(COLUMN_USER_NAME)
                if (columnIndex >= 0) { // Vérifiez que la colonne existe
                    it.getString(columnIndex)
                } else {
                    null
                }
            } else {
                null
            }
        }.also {
            db.close()
        }
    }
}